### PR TITLE
feat(api-client): import collection modal

### PR DESCRIPTION
**Pull request by hanspagel**
This PR adds an import modal that comes up if you pass an URL and a title (optional) as query parameters.

I’d prefer to disable this feature and merge, to then ff in a separate PR with an improved Electron integration.

Note: The feature will be disabled until we’ve gone through the UI.

**Preview**
Screenshot 2024-10-01 at 16 08 14

### CODE REVIEW

**Comment by antlio** [comment]
great work! found a small overflow issue on errors:
image

File: packages/api-client/src/components/ImportCollection/utils/isUrl.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/** Checks whether the given string is an URL */
+export function isUrl(input: string | null) {
+  return input && (input.startsWith('http://') || input.startsWith('https://'))
+}
```
Comment (antlio) [inline packages/api-client/src/components/ImportCollection/utils/isUrl.ts:4]: would it make send to use a url regex as a checker in the regexHelpers?

File: packages/api-client/src/components/ImportCollection/ImportNowButton.vue [hunk 2]
```diff
@@ -0,0 +1,101 @@
+<script lang="ts" setup>
+import { isDocument } from '@/components/ImportCollection/utils/isDocument'
```
Comment (antlio) [inline packages/api-client/src/components/ImportCollection/ImportNowButton.vue:2]: could this be removed? doesn't seem used here

File: packages/api-client/src/components/ImportCollection/utils/isUrl.ts (hunk 1 above)
Comment (hanspagel) [reply packages/api-client/src/components/ImportCollection/utils/isUrl.ts:4]: Hm … I think it’s fine. Way easier to read than any regex and not performance relevent.

**Review by marclave — APPROVED** [review]
@hanspagel let's merge after it's disabled? :) 

awesome work 🆒

File: packages/api-client/src/layouts/App/ApiClientApp.vue [hunk 3]
```diff
@@ -60,6 +62,8 @@ const fontsStyleTag = computed(
 )
 </script>
 <template>
+  <!-- Listen for paste and drop events, and look for `url` query parameters to import collections -->
+  <!-- <ImportCollectionListener> -->
```
Comment (hanspagel) [inline packages/api-client/src/layouts/App/ApiClientApp.vue:66]: disabled already

**Pull request merged**